From 24f0b4511f67a6a1dfc38c6a0549167ac2dedd02 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 17 Oct 2017 18:37:13 -0700
Subject: [PATCH] Fix build with gcc-7

Fixes warnings about fall through in case statements

../../git/transport/owr_transport_agent.c: In function 'bus_call':
../../git/transport/owr_transport_agent.c:393:20: error: this statement may fall through [-Werror=implicit-fallthrough=]
         is_warning = TRUE;

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 local/owr_local_media_source.c  | 1 +
 local/owr_media_renderer.c      | 2 +-
 local/owr_uri_source_agent.c    | 1 +
 transport/owr_transport_agent.c | 2 +-
 4 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/local/owr_local_media_source.c b/local/owr_local_media_source.c
index 9fd54d4..366151e 100644
--- a/local/owr_local_media_source.c
+++ b/local/owr_local_media_source.c
@@ -379,6 +379,7 @@ static gboolean bus_call(GstBus *bus, GstMessage *msg, gpointer user_data)
 
     case GST_MESSAGE_WARNING:
         is_warning = TRUE;
+        /* fallthru */
 
     case GST_MESSAGE_ERROR:
         if (is_warning) {
diff --git a/local/owr_media_renderer.c b/local/owr_media_renderer.c
index 5d688e8..84d95ac 100644
--- a/local/owr_media_renderer.c
+++ b/local/owr_media_renderer.c
@@ -179,7 +179,7 @@ static gboolean bus_call(GstBus *bus, GstMessage *msg, gpointer user_data)
 
     case GST_MESSAGE_WARNING:
         is_warning = TRUE;
-
+        /* fallthru */
     case GST_MESSAGE_ERROR:
         if (is_warning) {
             message_type = "Warning";
diff --git a/local/owr_uri_source_agent.c b/local/owr_uri_source_agent.c
index c9484d8..9eed5b7 100644
--- a/local/owr_uri_source_agent.c
+++ b/local/owr_uri_source_agent.c
@@ -165,6 +165,7 @@ static gboolean bus_call(GstBus *bus, GstMessage *msg, gpointer user_data)
 
     case GST_MESSAGE_WARNING:
         is_warning = TRUE;
+        /* fallthru */
 
     case GST_MESSAGE_ERROR:
         if (is_warning) {
diff --git a/transport/owr_transport_agent.c b/transport/owr_transport_agent.c
index 89d348b..78e323c 100644
--- a/transport/owr_transport_agent.c
+++ b/transport/owr_transport_agent.c
@@ -391,7 +391,7 @@ static gboolean bus_call(GstBus *bus, GstMessage *msg, gpointer user_data)
 
     case GST_MESSAGE_WARNING:
         is_warning = TRUE;
-
+        /* fallthru */
     case GST_MESSAGE_ERROR:
         if (is_warning) {
             message_type = "Warning";
-- 
2.14.2

